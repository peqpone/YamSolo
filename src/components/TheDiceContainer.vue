<script setup lang="ts">
import { computed } from 'vue';
import RenderDie from '@/components/RenderDie.vue';
import useDiceStore from '@/stores/dice';

const diceStore = useDiceStore();
const rawDice = computed(() => diceStore.rawDice);

const randomRotation = () => Math.floor(Math.random() * (10 + 10) - 10);
const style = () => ({ transform: `rotate(${randomRotation()}deg)` });

const toggleLock = (index:number) => {
  diceStore.toggleLock(index);
};

const isLocked = (die:Die) => die.isLocked;

</script>

<template>
  <div class="dice-area">
    <template
      v-for="(die, index) in rawDice"
      :key="index"
    >
      <render-die
        v-if="!isLocked(die)"
        :die-value="die.value"
        :style="style()"
        :class="`die-${index}`"
        @click="toggleLock(index)"
      />
    </template>
  </div>
  <div class="locked-dice-area">
    <template
      v-for="(die, index) in rawDice"
      :key="index"
    >
      <render-die
        v-if="isLocked(die)"
        :die-value="die.value"
        class="selected`"
        :class="`die-${index}`"
        @click="toggleLock(index)"
      />
    </template>
  </div>
</template>

<style scoped lang="scss">
.selected {
  width: 15%;
  margin: 3%;
}
</style>
