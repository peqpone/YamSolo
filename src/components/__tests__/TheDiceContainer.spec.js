import {
  describe, it, expect, beforeAll, vi,
} from 'vitest';

import { shallowMount } from '@vue/test-utils';
import { createPinia, setActivePinia } from 'pinia';
import TheDiceContainer from '../TheDiceContainer.vue';
import useDiceStore from '../../stores/dice';

describe('TheDiceContainer', () => {
  let diceStore;
  beforeAll(() => {
    setActivePinia(createPinia());
    Math.random = vi.fn(() => 1);
    diceStore = useDiceStore();
  });
  describe('Integration test', () => {
    let wrapper;
    it('should correctly mount initial state', () => {
      diceStore.$patch({
        rawDice: [
          { value: 1, isLocked: false },
          { value: 1, isLocked: false },
          { value: 1, isLocked: false },
          { value: 1, isLocked: false },
          { value: 1, isLocked: false },
        ],
      });
      wrapper = shallowMount(TheDiceContainer);
      expect(wrapper.html()).toMatchSnapshot();
    });
    it('click on a die should lock it', async () => {
      const die = wrapper.find('[dieid="0"]');
      await die.trigger('click');
      expect(diceStore.rawDice[0].isLocked).toBe(true);
    });
  });
});
