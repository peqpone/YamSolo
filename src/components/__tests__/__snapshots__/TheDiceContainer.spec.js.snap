// Vitest Snapshot v1

exports[`TheDiceContainer > Integration test > should correctly mount initial state 1`] = `
"<div class=\\"dice-area\\" data-v-1c9fb7f4=\\"\\">
  <render-die-stub dievalue=\\"1\\" style=\\"transform: rotate(10deg);\\" class=\\"die-0\\" data-v-1c9fb7f4=\\"\\"></render-die-stub>
  <render-die-stub dievalue=\\"1\\" style=\\"transform: rotate(10deg);\\" class=\\"die-1\\" data-v-1c9fb7f4=\\"\\"></render-die-stub>
  <render-die-stub dievalue=\\"1\\" style=\\"transform: rotate(10deg);\\" class=\\"die-2\\" data-v-1c9fb7f4=\\"\\"></render-die-stub>
  <render-die-stub dievalue=\\"1\\" style=\\"transform: rotate(10deg);\\" class=\\"die-3\\" data-v-1c9fb7f4=\\"\\"></render-die-stub>
  <render-die-stub dievalue=\\"1\\" style=\\"transform: rotate(10deg);\\" class=\\"die-4\\" data-v-1c9fb7f4=\\"\\"></render-die-stub>
</div>"
`;
